### PR TITLE
fix: Don't allow group chair to change group parent

### LINKED ISSUE #6037: Group edit form destroys group parent for rswg chairs
When an rswg chair edits the group (to add extresources for example), the control for parent is present (it should not be) with the only choice available set to "None".

### PR DESCRIPTION
**Pull request by pselkirk**
Fixes #6037

### CODE REVIEW

File: ietf/group/tests_info.py [hunk 1]
```diff
@@ -950,6 +950,31 @@ def test_edit_description_field(self):
...
+    def test_edit_parent_field(self):
+        group = GroupFactory(acronym='mars', parent=Group.objects.get(acronym='farfut'))
+        RoleFactory(group=group, name_id='chair')
+        url = urlreverse('ietf.group.views.edit', kwargs={'acronym': group.acronym, 'action': 'edit', 'field': 'parent'})
+
+        # parent is not shown to group chair
+        login_testing_unauthorized(self, 'mars-chair', url)
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(len(r.content), 0)
```
Comment (rjsparks) [inline ietf/group/tests_info.py:962]: It's not clear what the intent of this test block is? The view that would only edit the parent field should probably just refuse entry?

What about testing the form that allows editing the whole group (when you don't pass in field to the view). - I would expect a test that shows that the parent field isn't present on that form.

**Review by rjsparks — CHANGES_REQUESTED** [review]

File: ietf/group/tests_info.py (hunk 1 above)
Comment (pselkirk) [reply ietf/group/tests_info.py:962]: For the chair, /edit/parent/ behaves exactly like /edit/xyzzy/ - it renders an edit page with no editable fields, just a Submit button and a Back button, which both lead back to the about page.

However, I bolluxed the user creation, and what's actually going on there is a redirection to the login page. 🤦 

But yeah, I can test the whole group form.

**Review by rjsparks — CHANGES_REQUESTED** [review]
Thanks for adjusting the tests. This is close to ready but has one more change that should be made.

Right now, in this branch, /group/rswg/edit/parent presents the following:

image

There is risk of unexpected happening with submitting a null-form to the current code (which hunts for fields).

I suspect the current code would also let someone provide the parent field manually here (or on the full edit form) in a post and the value would be honored.

Please test those posts (attempting to change parent as a chair should result in no change).

I think the form view at /group/rswg/edit/parent should 403 unless you have the roles you've already identified for showing the field.

**Comment by pselkirk** [comment]
It turns out that the form is accepted (form.is_valid==True), but form.cleaned_data only contains the fields that are shown to the user - in the case of editing just the parent field, it's an empty dict.

**Comment by rjsparks** [comment]
I still think it's the right time to change the view behavior to 403 if there's no field to edit. (such as with your xyzzy example).

**Comment by pselkirk** [comment]
Unit testing on github is failing with `'ietf/name/models.py:104: error: Need type annotation for "next_states"  [var-annotated]'`. Did something change in the settings?

**Review by rjsparks — APPROVED** [review]

**Pull request merged**